sdk/tests/js/tests/tex-image-and-sub-image-2d-with-image-bitmap-utils.js

```diff
@@ -0,0 +1,169 @@
+    }
+
+    var width = gl.canvas.width;
+    var halfWidth = Math.floor(width / 2);
+    var quaterWidth = Math.floor(halfWidth / 2);
+    var height = gl.canvas.height;
+    var halfHeight = Math.floor(height / 2);
+    var quaterHeight = Math.floor(halfHeight / 2);
+
+    var top = flipY ? quaterHeight : (height - halfHeight + quaterHeight);
+    var bottom = flipY ? (height - halfHeight + quaterHeight) : quaterHeight;
+
+    var tl = redColor;
+    var tr = premultiplyAlpha ? ((optionsVal.alpha == 0.5) ? halfRed : (optionsVal.alpha == 1) ? redColor : blackColor) : redColor;
+    var bl = greenColor;
+    var br = premultiplyAlpha ? ((optionsVal.alpha == 0.5) ? halfGreen : (optionsVal.alpha == 1) ? greenColor : blackColor) : greenColor;
+
+    var loc;
+    if (bindingTarget == gl.TEXTURE_CUBE_MAP) {
+        loc = gl.getUniformLocation(program, "face");
+    }
+
+    var tolerance = 10;
+    for (var tt = 0; tt < targets.length; ++tt) {
+        if (bindingTarget == gl.TEXTURE_CUBE_MAP) {
+            gl.uniform1i(loc, targets[tt]);
+        }
+        // Draw the triangles
+        wtu.clearAndDrawUnitQuad(gl, [0, 0, 0, 255]);
+
+        // Check the top pixel and bottom pixel and make sure they have
+        // the right color.
+        debug("Checking " + (flipY ? "top" : "bottom"));
+        wtu.checkCanvasRect(gl, quaterWidth, bottom, 2, 2, tl, "shouldBe " + tl);
+        wtu.checkCanvasRect(gl, halfWidth + quaterWidth, bottom, 2, 2, tr, "shouldBe " + tr, tolerance);
+        debug("Checking " + (flipY ? "bottom" : "top"));
+        wtu.checkCanvasRect(gl, quaterWidth, top, 2, 2, bl, "shouldBe " + bl);
+        wtu.checkCanvasRect(gl, halfWidth + quaterWidth, top, 2, 2, br, "shouldBe " + br, tolerance);
+    }
+}
+
+function runTestOnBindingTargetImageBitmap(bindingTarget, program, bitmaps, optionsVal,
+    internalFormat, pixelFormat, pixelType, gl, tiu, wtu)
+{
+    var cases = [
+        { sub: false, bitmap: bitmaps.noFlipYPremul, flipY: false, premultiply: true },
+        { sub: true, bitmap: bitmaps.noFlipYPremul, flipY: false, premultiply: true },
+        { sub: false, bitmap: bitmaps.noFlipYUnpremul, flipY: false, premultiply: false },
+        { sub: true, bitmap: bitmaps.noFlipYUnpremul, flipY: false, premultiply: false },
+        { sub: false, bitmap: bitmaps.flipYPremul, flipY: true, premultiply: true },
+        { sub: true, bitmap: bitmaps.flipYPremul, flipY: true, premultiply: true },
+        { sub: false, bitmap: bitmaps.flipYUnpremul, flipY: true, premultiply: false },
+        { sub: true, bitmap: bitmaps.flipYUnpremul, flipY: true, premultiply: false },
+    ];
+
+    for (var i in cases) {
+        runOneIterationImageBitmapTest(cases[i].sub, bindingTarget, program, cases[i].bitmap,
+            cases[i].flipY, cases[i].premultiply, optionsVal, internalFormat, pixelFormat, pixelType, gl, tiu, wtu);
+    }
+}
+
+function runImageBitmapTest(bitmaps, alphaVal, internalFormat, pixelFormat, pixelType, gl, tiu, wtu)
```

Comment:
Could you rename this to "runImageBitmapTestImpl" or "runImageBitmapTestInternal" or something similar to indicate it's non-public?